azure/components/apigateway/setup.ftl

```diff
@@ -51,98 +152,102 @@
-    [#list apis as apiId, apiResources]
-
-        [#-- merge CMDB publisher contact over existing any existing details --]
-        [#if publishers[apiId].Enabled!false]
-            [#local apiManagementServicePublisher =
-                mergeObjects(
-                    apiManagementServicePublisher,
-                    publishers[apiId]
-                )
-            ]
+        [#-- API Management contact details. CMDB > openapi spec. --]
+        [#if openapiDefinition.info?has_content]
+            [#if openapiDefinition.info.contact?has_content]
+                [#local service = mergeObjects(
+                    service,
+                    {
+                        "ContactName" : openapiDefinition.info.contact.name!"",
+                        "ContactEmail" : openapiDefinition.info.contact.email!""
+                    }
+                )]
+            [/#if]
         [/#if]
-        
-        [#local api = apiResources["api"]]
-        [#local schema = apiResources["schema"]]
+        [#local service = mergeObjects(
+            service,
+            {} +
+            attributeIfContent("ContactName", solution.azure\:Contact.Name!"") +
+            attributeIfContent("ContactEmail", solution.azure\:Contact.Email!"")
+        )]
+
+        [#-- Extend Spec with extensions.                                        --]
+        [#-- Azure only supports two openapi extensions, all others are skipped. --]
+        [#-- Extensions: x-ms-paths & x-servers                                  --]
+        [#local extendedDefinition = 
+            azExtendOpenapiDefinition(
+                openapiDefinition,
+                openapiIntegrations,
+                openapiContext,
+                true
+            )]
+
+        [#list identityproviders?values as identityprovider]
+            [@createApiManagementServiceIdentityProvider
+                id=identityprovider.Id
+                name=identityprovider.Name
+                clientId=identityprovider.ObjectId
+                keyvaultId=keyvault
+                keyvaultSecret=identityprovider.SecretId
+                dependsOn=[service.Reference]
+            /]
+        [/#list]
 
         [@createApiManagementServiceApi
             id=api.Id
             name=api.Name
+            value=extendedDefinition
             path="test"
             dependsOn=[service.Reference]
         /]
 
-        [@createApiManagementServiceApiSchema
-            id=schema.Id
-            name=schema.Name
-            contentType="application/vnd.oai.openapi.components+json"
+        [@createApiManagementServiceProduct
+            id=product.Id
+            name=product.Name
             dependsOn=[api.Reference]
         /]
 
-        [#local productDependencies += [api.Reference]]
-
-    [/#list]
+        [@createApiManagementService
+            id=service.Id
+            name=service.Name
+            location=regionId
+            skuName="Developer"
```

Comment:
Should this be configurable?

Reply:
I have addressed this with the implementation of skuProfiles, in https://github.com/hamlet-io/engine#1305 & 14c3904